### PR TITLE
Core RC2

### PR DESCRIPTION
**Pull request by yvesgoeleven**
- Update ASB to Core RC2
- Fix the issue of message pumps (main input queue pump, that has to be reused between `SubscriptionManager` and `MessagePump`, and satellite input queues pumps)

### CODE REVIEW

**Comment by SeanFeldman** [comment]
@Particular/azure-service-bus-maintainers please review

File: src/Transport/Config/AzureServiceBusTransport.cs [hunk 1]
```diff
@@ -28,7 +29,7 @@ public override TransportInfrastructure Initialize(SettingsHolder settings, stri
 
             SetConnectivityMode(settings);
 
-            return new AzureServiceBusTransportInfrastructure(topology, settings);
+            return new AzureServiceBusTransportInfrastructure(topology, settings, settings.Get<SatelliteTransportAddressCollection>());
```
Comment (yvesgoeleven) [inline src/Transport/Config/AzureServiceBusTransport.cs]: why inject the collection if the settings holder is already injected?

File: src/Transport/Config/DefaultConfigurationValues.cs [hunk 2]
```diff
@@ -18,9 +19,16 @@ public SettingsHolder Apply(SettingsHolder settings)
             ApplyDefaultValuesForRules(settings);
             ApplyDefaultValuesForSerialization(settings);
             ApplyDefaultValuesForBrokeredMessageConventions(settings);
+            ApplyDefaultValuesForSatelliteTransportAddresses(settings);
             return settings;
         }
 
+        static void ApplyDefaultValuesForSatelliteTransportAddresses(SettingsHolder settings)
+        {
+            // TODO: review if we can improve this
```
Comment (yvesgoeleven) [inline src/Transport/Config/DefaultConfigurationValues.cs]: you could check on consumption if the settings key is configured

File: src/Transport/Seam/MessagePump.cs [hunk 3]
```diff
@@ -17,28 +19,19 @@ class MessagePump : IPushMessages, IDisposable
         RepeatedFailuresOverTimeCircuitBreaker circuitBreaker;
         ILog logger = LogManager.GetLogger(typeof(MessagePump));
         string inputQueue;
+        SatelliteTransportAddressCollection satelliteTransportAddresses;
 
         public MessagePump(ITopologySectionManager topologySectionManager, ITransportPartsContainer container)
         {
             this.topologySectionManager = topologySectionManager;
             this.container = container;
+            satelliteTransportAddresses = container.Resolve<ReadOnlySettings>().Get<SatelliteTransportAddressCollection>();
```
Comment (yvesgoeleven) [inline src/Transport/Seam/MessagePump.cs]: wouldn't it be better to inject ReadOnlySettings instead of querying the container here?

**Comment by yvesgoeleven** [comment]
@SeanFeldman added a few comments, but besides that looks good

File: src/Transport/Config/DefaultConfigurationValues.cs (hunk 2 above)
Comment (andreasohlund) [inline src/Transport/Config/DefaultConfigurationValues.cs]: I think the current way is good enough for now. I'll remove the todo?

**Pull request merged**